pandas/io/parsers.py

```diff
@@ -2165,6 +2231,129 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
         return values
 
 
+class BytesIOWrapper:
```

Comment:
this should be in pandas/io/common.py; though don't we already have something similar?

Reply:
xref https://github.com/pandas-dev/pandas/pull/37639#issuecomment-742278895 I checked with @twoertwein and we don't

Planning to move this to pandas/io/common.py and integrate it into `pandas.io.common.get_handle` to be consistent with what we do for other IO methods. 

xref https://github.com/pandas-dev/pandas/pull/37639#issuecomment-742650316 Might spin it off into a precursor PR since `BytesIOWrapper` is in principle useful outside the pyarrow reader